api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -88,6 +90,19 @@
   String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
   boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
 
+  /**
+   * Forces the compaction order based on the value.
+   * <p>
+   * If rewrite.job-order=bytes, then compact based on ascending order of partition size in bytes.
```

Comment:
order of the **job group**

Reply:
Just because it isn't necessarily a full partition

Reply:
Maybe to make this clearer,

bytes : Rewrite the smallest job groups first
files: Rewrite the job groups with the most files first
none: Rewrite job groups in the order they were planned (no specific ordering)

Reply:
When I commented on the size before I wasn't sure which made more sense, do these sort orders make sense for your use case? I was generally thinking about performance and how usually we want to leave possible stragglers till the end but I could be wrong. I'm interested to hear your reasoning and why we should go one way or the other.

Reply:
I think these can be the use cases (I can be convinced otherwise):
- bytes-ascending (avoid larger ones as they are more likely to be stragglers).
- bytes-descending (compact groups with more bytes to gain performance, customer can enable `partial-progress.enabled` and set a low max commit to compact the bulkiest group greedily. This should be done at a maintenance time window where stragglers are not a problem).
- files-descending (intention would be to reduce number of files to gain performance).

I not sure about the files-ascending use case. I feel we still might be missing on many other customer use cases. 

Thought: Iceberg can have some pre-defined rewrite job orders (initial PR scope). Can we think of introducing a custom order where the user gets to pass the comparator at runtime and we load it via reflection?

Reply:
I would probably just add files ascending as well just for symmetry. I think we should avoid any reflection based solutions. We should keep the API as tight as we can, if there are active use cases for other patterns later we can address them then.

Reply:
Thanks, for the feedback. I'll add the following job orders: bytes-asc, bytes-desc, files-asc, files-desc, none.